examples/neighbors/approximate_nearest_neighbors.py

```diff
@@ -178,134 +100,221 @@ def _transform(self, X):
     """Load MNIST, shuffle the data, and return only n_samples."""
     mnist = fetch_openml("mnist_784", as_frame=False, parser="pandas")
     X, y = shuffle(mnist.data, mnist.target, random_state=2)
     return X[:n_samples] / 255, y[:n_samples]
 
 
-def run_benchmark():
-    datasets = [
-        ("MNIST_2000", load_mnist(n_samples=2000)),
-        ("MNIST_10000", load_mnist(n_samples=10000)),
-    ]
-
-    n_iter = 500
-    perplexity = 30
-    metric = "euclidean"
-    # TSNE requires a certain number of neighbors which depends on the
-    # perplexity parameter.
-    # Add one since we include each sample as its own neighbor.
-    n_neighbors = int(3.0 * perplexity + 1) + 1
-
-    tsne_params = dict(
-        init="random",  # pca not supported for sparse matrices
-        perplexity=perplexity,
-        method="barnes_hut",
-        random_state=42,
-        n_iter=n_iter,
-        learning_rate="auto",
-    )
-
-    transformers = [
-        ("AnnoyTransformer", AnnoyTransformer(n_neighbors=n_neighbors, metric=metric)),
-        (
-            "NMSlibTransformer",
-            NMSlibTransformer(n_neighbors=n_neighbors, metric=metric),
+# %%
+# We benchmark the different exact/approximate nearest neighbors transformers.
+import time
+
+from sklearn.manifold import TSNE
+from sklearn.neighbors import KNeighborsTransformer
+from sklearn.pipeline import make_pipeline
+
+datasets = [
+    ("MNIST_10000", load_mnist(n_samples=10_000)),
+    ("MNIST_20000", load_mnist(n_samples=20_000)),
+]
+
+n_iter = 500
+perplexity = 30
+metric = "euclidean"
+# TSNE requires a certain number of neighbors which depends on the
+# perplexity parameter.
+# Add one since we include each sample as its own neighbor.
+n_neighbors = int(3.0 * perplexity + 1) + 1
+
+tsne_params = dict(
+    init="random",  # pca not supported for sparse matrices
+    perplexity=perplexity,
+    method="barnes_hut",
+    random_state=42,
+    n_iter=n_iter,
+    learning_rate="auto",
+)
+
+transformers = [
+    (
+        "KNeighborsTransformer",
+        KNeighborsTransformer(n_neighbors=n_neighbors, mode="distance", metric=metric),
+    ),
+    (
+        "NMSlibTransformer",
+        NMSlibTransformer(n_neighbors=n_neighbors, metric=metric),
+    ),
+    (
+        "PyNNDescentTransformer",
+        PyNNDescentTransformer(
+            n_neighbors=n_neighbors, metric=metric, parallel_batch_queries=True
         ),
-        (
-            "KNeighborsTransformer",
+    ),
+]
+
+for dataset_name, (X, y) in datasets:
+
+    msg = "Benchmarking on %s:" % dataset_name
+    print("\n%s\n%s" % (msg, "-" * len(msg)))
+
+    for transformer_name, transformer in transformers:
+        longest = np.max([len(name) for name, model in transformers])
+        whitespaces = " " * (longest - len(transformer_name))
+        start = time.time()
+        transformer.fit(X)
+        fit_duration = time.time() - start
+        print("%s: %s%.3f sec (fit)" % (transformer_name, whitespaces, fit_duration))
```

Comment:
Also, instead of computing the longest name to dynamically adjust the number of whitespaces, let's fix a maximum length (e.g. 40) and use the `:<40` formatter such as in the following example:

```
>>> texts = ["short text", "longer text", "very long text"]
>>> 
>>> for text, number in zip(texts, range(3)):
...     print(f"{text:<40} {number:.2f}")
... 
short text                               0.00
longer text                              1.00
very long text                           2.00
```

You can drop the `:` to make the printing logic simpler.